Wikipedia/Code/AccountCreationViewController.swift

```diff
@@ -0,0 +1,285 @@
+        if (textField == usernameField) {
+            passwordField.becomeFirstResponder()
+        } else if (textField == passwordField) {
+            passwordRepeatField.becomeFirstResponder()
+        } else if (textField == passwordRepeatField) {
+            emailField.becomeFirstResponder()
+        } else {
+            assert(((textField == emailField) || (textField == captchaViewController?.captchaTextBox)), "Received -textFieldShouldReturn for unexpected text field: \(textField)")
+            save()
+        }
+        return true
+    }
+
+    fileprivate func enableProgressiveButton(_ enabled: Bool) {
+        navigationItem.rightBarButtonItem?.isEnabled = enabled
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        enableProgressiveButton(false)
+        WMFAlertManager.sharedInstance.dismissAlert()
+        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UITextFieldTextDidChange, object: captchaViewController?.captchaTextBox)
+        super.viewWillDisappear(animated)
+    }
+    
+    fileprivate var showCaptchaContainer: Bool = false {
+        didSet {
+            rightButton?.title = showCaptchaContainer ? localizedStringForKeyFallingBackOnEnglish("button-done") : localizedStringForKeyFallingBackOnEnglish("button-next")
+            let duration: TimeInterval = 0.5
+            view.setNeedsUpdateConstraints()
+            
+            if showCaptchaContainer {
+                captchaViewController?.captchaTextBox.perform(#selector(becomeFirstResponder), with: nil, afterDelay: 0.4)
+                funnel?.logCaptchaShown()
+                DispatchQueue.main.async(execute: {
+                    UIView.animate(withDuration: duration, animations: {
+                        self.captchaContainer.alpha = 1
+                        self.scrollView.scrollSubView(toTop: self.captchaContainer, animated:false)
+                    }, completion: {(completed: Bool) -> Void in
+                        self.enableProgressiveButton(false)
+                    })
+                })
+            }else{
+                DispatchQueue.main.async(execute: {
+                    WMFAlertManager.sharedInstance.dismissAlert()
+                    UIView.animate(withDuration: duration, animations: {
+                        self.captchaContainer.alpha = 0
+                        self.scrollView.setContentOffset(CGPoint.zero, animated: false)
+                    }, completion: {(completed: Bool) -> Void in
+                        self.captchaViewController?.captchaTextBox.text = ""
+                        self.captchaViewController?.captchaImageView.image = nil
+                        // Pretent a text field changed so the progressive button state gets updated.
+                        self.textFieldDidChange(nil)
+                    })
+                })
+            }
+        }
+    }
+    
+    private var _captchaUrl: URL? = nil
+    
+    fileprivate var captchaUrl: URL? {
```

Comment:
This can be replaced with a single `var` and a `didSet {`

```
fileprivate var captchaURL: URL? {
     didSet {
          guard let newURL = captchaURL else {
               return;
          }
          refreshCaptchaImage()
     }
}
```

Also, URL should be all caps